compiler/rustc_hir_typeck/src/method/probe.rs

```diff
@@ -1981,6 +1987,46 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
         }
     }
 
+    /// Determine if the associated item withe the given DefId matches
+    /// the desired name via a doc alias.
+    fn matches_by_doc_alias(&self, def_id: DefId) -> bool {
+        if let Some(name) = self.method_name {
+            if let Some(local_def_id) = def_id.as_local() {
+                let hir_id = self.fcx.tcx.hir().local_def_id_to_hir_id(local_def_id);
+                let attrs = self.fcx.tcx.hir().attrs(hir_id);
+                for attr in attrs {
+                    if let sym::doc = attr.name_or_empty() {
+                        if let Some(values) = attr.meta_item_list() {
+                            for v in values {
+                                if v.name_or_empty().as_str() != "alias" {
```

Comment:
Could you add `alias` to the symbols table in `rustc_span/src/symbols.rs` (I think that was the path) instead of doing this string comparison?